autoscheduler/user_sessions/views.py

```diff
@@ -4,6 +4,47 @@
 from rest_framework.parsers import JSONParser
 from django.contrib.auth.models import User
 
+def _set_state_in_session(request, key: str):
+    """ Function that sets the given key in our session to the value of the key in the
+        request body. Must have term and the given key in the body of the request.
+        Used for save_courses and save_availabilities
+    """
+    try:
+        body = json.loads(request.body.decode())
+        objs = body.get(key, {})
+        term = body.get('term')
+        if not term:
+            return Response(f'Request body must contain {key} and term', status=400)
+    except (UnicodeError, json.JSONDecodeError):
+        return Response(status=400)
```

Comment:
Assuming you don't actually want to use the empty dict default for `objs` like I said in my above comment, this should look like the following (since request data is already processed by `JSONParser`). Also make sure you're setting `Content-Type: 'application/json'` in request headers.
```suggestion
    objs = request.data.get(key)
    term = request.data.get('term')
    if objs is None or term is None:
        return Response(f'Request body must contain {key} and term', status=400)
```